### PR TITLE
VM fails to start on M1 CPU when CPU cores set to more than the host has #1621

### PR DESCRIPTION
**Pull request by NotPranav0**
- returns an error from validate.go if --cpus is higher than hardware cpus
- wasn't sure how to edit the config to automatically set --cpus to runtime.cpus

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Please squash commits

File: pkg/limayaml/validate.go [hunk 1]
```diff
@@ -108,6 +108,12 @@ func Validate(y LimaYAML, warn bool) error {
 		return errors.New("field `cpus` must be set")
 	}
 
+	if (*y.CPUs > runtime.NumCPU()) {
+		return fmt.Errorf("field `cpus` is set to %d, which is greater than the number of CPUs available (%d)", *y.CPUs, runtime.NumCPU())
+	}
+
+	
+
```
Comment (jandubois) [inline pkg/limayaml/validate.go:116]: Please add just a single empty line.

**Comment by AkihiroSuda** [comment]
@NotPranav0 Could you update the PR?

**Comment by jandubois** [comment]
Please squash your commits (there are currently 6 commits in your PR; there should be only one).

There is also a linter error because changes have not been formatted properly.

**Comment by NotPranav0** [comment]
> Please squash your commits (there are currently 6 commits in your PR; there should be only one).
> 
> There is also a linter error because changes have not been formatted properly.

**Comment by NotPranav0** [comment]
i think i messed up the squashing, ill make a new PR with a cleaner history

**Comment by jandubois** [comment]
> i think i messed up the squashing, ill make a new PR with a cleaner history

You don't need to create a new PR; you can clean up your branch locally and then use `git push -f` to overwrite the existing branch with your new version.

**Pull request closed without merging**